lxd/instances_post.go

```diff
@@ -404,6 +405,181 @@ func createFromMigration(s *state.State, r *http.Request, projectName string, pr
 	return operations.OperationResponse(op)
 }
 
+// createFromConversion uploads the image to the server's backup directory, converts it into the
+// the raw format, and imports the LXD instance using the converted image.
+func createFromConversion(s *state.State, r *http.Request, projectName string, profiles []api.Profile, req *api.InstancesPost) response.Response {
+	if s.DB.Cluster.LocalNodeIsEvacuated() && r.Context().Value(request.CtxProtocol) != "cluster" {
+		return response.Forbidden(fmt.Errorf("Cluster member is evacuated"))
+	}
+
+	// Validate migration mode.
+	if req.Source.Mode != "push" {
+		return response.NotImplemented(fmt.Errorf("Mode %q not implemented", req.Source.Mode))
+	}
+
+	// Parse the architecture name
+	architecture, err := osarch.ArchitectureId(req.Architecture)
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	dbType, err := instancetype.New(string(req.Type))
+	if err != nil {
+		return response.BadRequest(err)
+	}
+
+	// Currently, only VMs can be converted.
+	if dbType != instancetype.VM {
+		return response.BadRequest(fmt.Errorf("Instance type not supported %q", req.Type))
+	}
+
+	// Validate conversion options.
+	convertOpts := req.Source.ConversionOptions
+	for _, opt := range convertOpts {
+		if opt == "none" {
+			if len(convertOpts) > 1 {
+				return response.BadRequest(fmt.Errorf("Conversion option %q cannot be used in combination with other options", "none"))
```

Comment:
Why do we have `none` rather than having the client pass an explicitly empty list when they dont want any default conversions?

Reply:
Currently, both approaches are valid - using either `--conversion=` or `--conversion=none`. Will remove that flag.